ethereum_history_api/circuits/lib/src/rlp/types.nr

```diff
@@ -25,6 +25,11 @@ struct RlpFragment {
 }
 
 impl RlpFragment {
+
+    fn shift(self, shift: u64) -> RlpFragment {
+        RlpFragment { offset: self.offset + shift, length: self.length, data_type: self.data_type }
```

Comment:
This can cause the offset + length to be outside the data. Let's add an assertion & a test

Reply:
Also - let's rename to shift_right